src/_pytest/compat.py

```diff
@@ -20,6 +20,16 @@
 import attr
 import py
 
+# fmt: off
+# Workaround for https://github.com/sphinx-doc/sphinx/issues/10351.
+# If `overload` is imported from `compat` instead of from `typing`,
+# Sphinx doesn't recognize it as `overload` and the API docs for
+# overloaded functions look good again. But type checkers handle
+# it fine.
+# fmt: on
+if True:
+    from typing import overload as overload
```

Comment:
Curious: is the "import overload as overload" part of the workaround?

Reply:
This is the way to explicitly re-export a value in type checking (see https://mypy.readthedocs.io/en/stable/config_file.html?highlight=export#confval-implicit_reexport).